src/app/components/layouts/column-container/terra-two-columns-container.component.ts

```diff
@@ -37,11 +53,61 @@ export class TerraTwoColumnsContainerComponent
         this.rightColumn = this.leftRightColXS() + this.rightColMD() + this.rightColLG();
     }
 
-    constructor()
+    private readonly basePath:string;
+
+    constructor(private route:ActivatedRoute,
+                private router:Router)
     {
+        this.basePath = router.routerState.snapshot.url;
+        this.setColumnHidden('left');
+
+        let subscribable:Observable<Event> = this.router.events.filter((event:RouterEvent) =>
+        {
+            return event instanceof NavigationEnd && event.urlAfterRedirects.startsWith(this.basePath);
+        });
+
+        this.subscription = subscribable.subscribe((event:NavigationEnd) =>
+        {
+            this.leftColumn = this.leftRightColXS() + this.leftColMD() + this.leftColLG();
+            this.rightColumn = this.leftRightColXS() + this.rightColMD() + this.rightColLG();
+
+            if(event.url !== event.urlAfterRedirects)
+            {
+                this.setColumnHidden('right');
+            }
+            else
+            {
+                this.setColumnHidden('left');
+            }
+        });
+
         this.leftColumnWidth = this._leftColumnWidth; // trigger calculation for default values
     }
 
+    public ngOnInit():void
+    {
+        this.setColumnHidden('left');
```

Comment:
call this function either in the `constructor` or in `ngOnInit`, not both.

Reply:
done